torch/optim/optimizer.py

```diff
@@ -190,22 +275,22 @@ def __init__(self, params, defaults):
             param_groups = [{'params': param_groups}]
 
         for param_group in param_groups:
+            assert isinstance(param_group, dict)
```

Comment:
The very first thing that `add_param_group` does is throw an error if `not isinstance(param_group, dict)`, so I don't think this any side effects?

```python
    def add_param_group(self, param_group: Dict[str, Any]) -> None:
        ...
        if not isinstance(param_group, dict):
            raise TypeError(f"param_group must be a dict, but got {type(param_group)}")

```

I am performing the `assert` bc the lines above this:

```python
        if not isinstance(param_groups[0], dict):
            param_groups = [{'params': param_groups}]
```
do not actually convert `List[Tensor]` into `Dict[List[Tensor]]` since type checkers don't support this style of narrowing.

An equivalent assert would be `assert not isinstance(param_groups, torch.Tensor)`, but I was pretty sure this was safe bc of the immediately proceeding `TypeError` on the same condition.

Reply:
Would a cast accomplish the same thing here?

Reply:
yes